ncm-network/src/main/perl/network.pm

```diff
@@ -1409,29 +1412,34 @@ sub make_ifup
 #   best to also enable the network service with ncm-chkconfig
 sub enable_network_service
 {
-    my ($self) = @_;
+    my ($self, $allow_nm) = @_;
     # do not start it!
-    return $self->runrun([qw(/sbin/chkconfig --level 2345 network on)]);
+    if ($self->_is_executable($CHKCONFIG_CMD)) {
+        return $self->runrun([qw(/sbin/chkconfig --level 2345 network on)]);
+    } elsif (defined($allow_nm) && $allow_nm) {
+        return $self->runrun(['systemctl', 'enable', 'NetworkManager.service']);
+    } else {
+        $self->warn("chkconfig unavailable, NetworkManager not allowed");
+    }
 }
 
 # If allow is defined and false, disable and stop NetworkManager
 sub disable_networkmanager
 {
-    my ($self, $allow) = @_;
+    my ($self, $allow_nm) = @_;
 
     # allow NetworkMnager to run or not?
-    if (defined($allow) && !$allow) {
+    if (defined($allow_nm) && !$allow_nm) {
         # no checking, forcefully stopping NetworkManager
         # warning: this can cause troubles with the recovery to previous state in case of failure
         # it's always better to disable the NetworkManager service with ncm-chkconfig and have it run pre ncm-network
-        my @disablenm_cmds;
-
-        # TODO: do something smart with 'require NCM::Component::Systemd::...' to turn it off
-        push(@disablenm_cmds, [qw(/sbin/chkconfig --level 2345 NetworkManager off)]);
-
-        push(@disablenm_cmds, [CAF::Service->new(["NetworkManager"], log => $self), "stop"]);
-
-        $self->runrun(@disablenm_cmds);
+        if ($self->_is_executable($CHKCONFIG_CMD)) {
+            # TODO: do something smart with 'require NCM::Component::Systemd::...' to turn it off
+            return $self->runrun([qw(/sbin/chkconfig --level 2345 network off)]);
```

Comment:
why do you stop network, while the comments only mention networkmanager
and also, what does this have to do with chkconfig being present